web/src/user_profile.js

```diff
@@ -324,6 +352,26 @@ export function get_custom_profile_field_data(user, field, field_types) {
     return profile_field;
 }
 
+export function update_user_custom_profile_fields(user) {
+    if (!(modals.any_active() && modals.active_modal() === "#user-profile-modal")) {
+        return;
+    }
```

Comment:
You should also skip if it's open for a different user than the one updated.

Reply:
Done.